### ISSUE TITLE
Help needed for creating a Scraper for big7.com

### ISSUE THREAD

**Issue by FleX246** [stashapp/CommunityScripts#330]
Hey,

maybe someone can help me create a scraper for the website big7.com.

I have already learned how to create a scraper in general and already have a working scraper that reads (scrapes)
- the title
- the details
- the date

The tags are also read but not split into single tags.
Unfortunately the preview image doesn't work at all and I just can't get it to work.

My scraper is attached as a txt file.
[Big7_Scenes.txt](https://github.com/user-attachments/files/15775813/Big7_Scenes.txt)

Thank you for any help!

**Comment by Maista6969** [comment]
Sure! Here are a few points I thought of while reading through your scraper and testing it for myself.

As a first step I'd recommend writing your XPath selectors in terms of classes, IDs or even attributes instead of using the full absolute path: this doesn't always make them more stable, but it certainly makes them more readable for when you come back to the scraper later 😁

In this particular case the "useCDP" flag is not necessary: this makes scrapers use a headless browser to access the page and will only slow things down when it's not needed.

The tags appear on the page as separate elements, so we can actually select them as such and avoid having to split the string on commas by hand: I suggest the selector `//div[@class="video_categories"]/a` for this

When it comes to the image I actually found that I needed to combine two selectors to get good results: some scenes have a preview image in the HTML, like [Big Boy Play](https://www.big7.com/playvideo/404212.html), while others have it set as the `poster` attribute on the video element, like [Erste Hilfe Kurs](https://www.big7.com/playvideo/422423.html): I used the selector `//video/@poster | //div[@class="video_preview_image"]/img/@src` which will get both the poster and the preview image. I haven't found any pages that have BOTH of these elements, but when you use a selector that finds multiple results and Stash only needs one result it will use the first one so this will work either way.

This is the version of the scraper that I ended up with, please feel free to ask if you'd like me to clarify anything I've done here 🙂
[big7.txt](https://github.com/user-attachments/files/15776092/big7.txt)

**Comment by FleX246** [comment]
Wow that was super fast!
Thank you very much!!!

I think I understood your suggestion with the ids, classes, etc.

The website seems to be programmed a bit strangely.
If you are not logged in, a different publication date is displayed than when you are logged in.

As an example, with this link the scraper returns 2024-05-20 (not logged in), but I want 2019-10-08 (logged in)
[https://www.big7.com/playvideo/200437.html](url)

That's why I added the usecdp parameter again.
(I have set up a headless browser for MDH anyway.)

I also adapted your selectors according to your example, as they are apparently different when you are logged in.

As far as I can see, all of the scraper's functions now work.
(Only when logged in via headless browser)

This is my version now
Did I write the selector tags better this way? (Thanks again for your Examples)
[Big7_Scenes.txt](https://github.com/user-attachments/files/15777148/Big7_Scenes.txt)
